File: python_programming_basics/while_loops/l_06_max_number.py
command = input()
max_number = int(command)

while command != "Stop":
    number = int(command)

    if number > max_number:
        max_number = number

    command = input()

print(max_number)


# solution 2
# from math import inf
#
# command = input()
# max_number = -inf
#
# while command != "Stop":
#     current_number = int(command)
#
#     if current_number > max_number:
#         max_number = current_number
#
#     command = input()
#
# print(max_number)

File: JavaScript Basics/05. While Loop/l-06-Max-Number.js
function findMaxNumber(input) {
    let index = 0;
    let maxNumber = Number.NEGATIVE_INFINITY;

    while (input[index] !== "Stop") {
        let number = parseInt(input[index]);

        if (number > maxNumber) {
            maxNumber = number;
        }

        index++;
    }

    console.log(maxNumber);
}

findMaxNumber(["100", "99", "80", "70", "Stop"]);
findMaxNumber(["-10", "20", "-30", "Stop"]);
findMaxNumber(["45", "-20", "7", "99", "Stop"]);
findMaxNumber(["999", "Stop"]);
findMaxNumber(["-1", "-2", "Stop"]);
